adapters/impactify/impactify.go

```diff
@@ -0,0 +1,212 @@
+package impactify
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strings"
+)
+
+type ImpactifyAdapter struct {
+	endpoint string
+}
+
+type ImpactifyExtBidder struct {
+	Impactify openrtb_ext.ExtImpImpactify `json:"impactify"`
+}
+
+type DefaultExtBidder struct {
+	Bidder openrtb_ext.ExtImpImpactify `json:"bidder"`
+}
+
+func (a *ImpactifyAdapter) MakeRequests(bidRequest *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+
+	var adapterRequests []*adapters.RequestData
+	var referer string = ""
+
+	for i := 0; i < len(bidRequest.Imp); i++ {
+		ref, err := checkImp(&bidRequest.Imp[i])
+		// If the parsing is failed, remove imp and add the error.
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if referer == "" && ref != "" {
+			referer = ref
+		}
+		// Check if imp comes with bid floor amount defined in a foreign currency
+		if bidRequest.Imp[i].BidFloor > 0 && bidRequest.Imp[i].BidFloorCur != "" && strings.ToUpper(bidRequest.Imp[i].BidFloorCur) != "USD" {
+			// Convert to US dollars
+			convertedValue, err := reqInfo.ConvertCurrency(bidRequest.Imp[i].BidFloor, bidRequest.Imp[i].BidFloorCur, "USD")
+			if err != nil {
+				return nil, []error{err}
+			}
+			bidRequest.Imp[i].BidFloorCur = "USD"
+			bidRequest.Imp[i].BidFloor = convertedValue
+		}
+
+		var impactifyExt ImpactifyExtBidder
+		var defaultExt DefaultExtBidder
+		json.Unmarshal(bidRequest.Imp[i].Ext, &defaultExt)
+		impactifyExt.Impactify = defaultExt.Bidder
+		bidRequest.Imp[i].Ext, _ = json.Marshal(impactifyExt)
```

Comment:
We use it because our bidding server will handle ext params with "impactify" object. Because on our bidding we can manage another ext params. We are not promoting impactify, it's just for the request be handle by our bidding server.
I hope it's clear :)

Reply:
Ah, ok. You're essentially renaming the object then, not promoting it. Gotcha.